scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -2002,11 +2002,6 @@ def _check_properties(self):
                 _err(f"enum in {self.path} for property '{prop_name}' "
                      "is not a list")
 
-            if "const" in options and not isinstance(options["const"],
-                                                     (int, str)):
-                _err(f"const in {self.path} for property '{prop_name}' "
-                     "is not a scalar")
-
```

Comment:
This check is still needed. We wouldn't want to allow all types of arrays. For example, phandle-array would not make sense here. Please rework this to check for the expanded list of array type properties you really want to allow.

Reply:
@galak I'm confused about why you moved the check to a different function; can you explain in the commit log?

Reply:
It was a little easier to implement there and not duplicate code that was already in place in that function.  (but its been a few weeks so I forget exactly my reasoning ;).

Add a comment to the commit message.